pyth-sdk-solana/src/state.rs

```diff
@@ -737,4 +747,174 @@ mod test {
+                    size:           header.size,
+                    ptype:          match price_type {
+                        0 => PriceType::Unknown,
+                        1 => PriceType::Price,
+                        _ => unreachable!(),
+                    },
+                    expo:           exponent,
+                    num:            num_,
+                    num_qt:         num_qt_,
+                    last_slot:      last_slot_,
+                    valid_slot:     valid_slot_,
+                    ema_price:      Rational {
+                        val:   twap_.val_,
+                        numer: twap_.numer_,
+                        denom: twap_.denom_,
+                    },
+                    ema_conf:       Rational {
+                        val:   twac_.val_,
+                        numer: twac_.numer_,
+                        denom: twac_.denom_,
+                    },
+                    timestamp:      timestamp_,
+                    min_pub:        min_pub_,
+                    drv2:           unused_1_ as u8,
+                    drv3:           unused_2_ as u16,
+                    drv4:           unused_3_ as u32,
+                    prod:           product_account,
+                    next:           next_price_account,
+                    prev_slot:      prev_slot_,
+                    prev_price:     prev_price_,
+                    prev_conf:      prev_conf_,
+                    prev_timestamp: prev_timestamp_,
+                    agg:            PriceInfo::from(agg_),
+                    comp:           comp_.map(|comp| PriceComp {
+                        publisher: comp.pub_,
+                        agg:       PriceInfo::from(comp.agg_),
+                        latest:    PriceInfo::from(comp.latest_),
+                    }),
+                }
+            }
+        }
+
+        impl From<pyth_oracle::MappingAccount> for MappingAccount {
+            fn from(other: pyth_oracle::MappingAccount) -> Self {
+                let pyth_oracle::MappingAccount {
+                    header,
+                    number_of_products,
+                    unused_,
+                    next_mapping_account,
+                    products_list,
+                } = other;
+
+                Self {
+                    magic:    header.magic_number,
+                    ver:      header.version,
+                    atype:    header.account_type,
+                    size:     header.size,
+                    num:      number_of_products,
+                    unused:   unused_,
+                    next:     next_mapping_account,
+                    products: products_list,
+                }
+            }
+        }
+
+        // We then compare the type sizes to get some guarantee that the layout is likely the same.
+        // Combined with the isomorphism above we should be able to catch layout changes that were
+        // not propogated from the pyth-client program.
```

Comment:
propagated

Reply:
Thanks, will fix, along with `definitinos` in the above comment as well after I clarify it a bit better.